packages/Confirmation/src/Confirmation.js

```diff
@@ -0,0 +1,147 @@
+  confirmHeaderStyles,
+  confirmDescriptionStyles,
+  confirmFooterStyles,
+} from "./Confirmation.styles";
+
+const propTypes = {
+  buttonSize: PropTypes.oneOf(ShirtSizes.DEFAULT),
+  confirmButtonType: PropTypes.oneOf(["primary", "destructive"]),
+  confirmLabel: PropTypes.node.isRequired,
+  description: PropTypes.node,
+  heading: PropTypes.node.isRequired,
+  defaultIsOpen: PropTypes.bool,
+  isPending: PropTypes.bool,
+  onCancel: PropTypes.func.isRequired,
+  onClose: PropTypes.func,
+  onConfirm: PropTypes.func.isRequired,
+  /** Render prop for rendering the trigger element that toggles the render panel */
+  renderTrigger: PropTypes.func,
+};
+
+const defaultProps = {
+  buttonSize: ShirtSizes.MEDIUM,
+  confirmButtonType: "destructive",
+  description: null,
+  defaultIsOpen: false,
+  isPending: false,
+  renderTrigger: null,
+  onClose: () => {},
+};
+
+const Confirmation = props => {
+  const [isConfirmOpen, setIsConfirmOpen] = React.useState(false);
+  const {
+    heading,
+    buttonSize,
+    confirmButtonType,
+    confirmLabel,
+    description,
+    defaultIsOpen,
+    isPending,
+    onConfirm,
+    onCancel,
+    onClose,
+    ...moreProps
+  } = props;
+  const confirmButtonRef = React.useRef(null);
+  const triggerRef = React.useRef(null);
+  const confirmPanelRefId = React.useRef(null);
+
+  const focusConfirmButton = () => {
+    if (confirmButtonRef.current) confirmButtonRef.current.focus();
+  };
+
+  const handleOpenConfirm = () => {
+    setIsConfirmOpen(true);
+    setTimeout(() => {
+      focusConfirmButton();
+    }, 200);
+  };
+
+  React.useEffect(() => {
+    if (defaultIsOpen && !isConfirmOpen) {
+      handleOpenConfirm();
+    }
+  }, [confirmButtonRef, defaultIsOpen]);
+
+  const triggerProps = {
+    isConfirmOpen,
+    handleOpenConfirm,
+  };
+
+  const handleOnConfirm = () => {
+    setIsConfirmOpen(false);
+    onConfirm();
+  };
+
+  const handleOnCancel = () => {
+    setIsConfirmOpen(false);
+    onCancel();
+  };
+
+  const handleCloseConfirm = () => {
+    setIsConfirmOpen(false);
+    if (triggerRef.current) triggerRef.current.focus();
+    onClose();
+  };
+
+  const renderTrigger = () => {
+    const triggerComponent = props.renderTrigger(triggerProps);
+    // wrapping the returned item in a function to avoid needing to tab twice
+    // https://github.com/acl-services/paprika/issues/126
+    return () =>
+      React.cloneElement(triggerComponent, {
+        triggerRef,
+        confirmPanelRefId: confirmPanelRefId.current,
+      });
+  };
+
+  const I18n = useI18n();
+  // need to add moreProps to the popover
+  return (
+    <Popover isOpen={isConfirmOpen} onClose={handleCloseConfirm} {...moreProps}>
+      {props.renderTrigger ? <Popover.Trigger>{renderTrigger()}</Popover.Trigger> : null}
+      <Popover.Content id={confirmPanelRefId.current}>
+        <div css={contentStyles}>
+          <div css={confirmStyles}>
+            <div css={confirmHeaderStyles}>{heading}</div>
```

Comment:
As per my previous suggestion to make `heading` a `string` and wrap it in a `<Header>`... that would mean no need for `confirmHeaderStyles`.